sdk/core/azure-core/src/context.cpp

```diff
@@ -5,14 +5,14 @@
 
 using namespace Azure::Core;
 
-const Context Context::ApplicationContext;
+Context Context::ApplicationContext{Context::CreateNewRoot()};
```

Comment:
I get why this is, but it still bugs me to have the ApplicationContext NOT be const....  Replacing it feels like a potential for race condition frees

Reply:
This initialization happens before anything else executes, there can be no races, by definition (literally - the language requires that there be no races in static variable initialization).

Reply:
And if ApplicationContext is const, it means you cannot cancel ApplicationContext (because cancellation is a mutation of the context's shared state).

An immutable ApplicationContext, means that the only way to cancel ApplicationContext is to create a copy of ApplicationContext and then cancel the copy.

Which feels REALLY awkward to me.
 
Can you help me understand why a non-const ApplicationContext bugs you?